aspnet/tutorials/dotnet-watch.rst

```diff
@@ -0,0 +1,198 @@
+  log  : Installing Microsoft.DotNet.Watcher.Core 1.0.0-preview2-final.
+  log  : Installing Microsoft.DotNet.Watcher.Tools 1.0.0-preview2-final.
+
+Running ``dotnet`` commands using ``dotnet watch``
+--------------------------------------------------
+
+Any ``dotnet`` command can be run through ``dotnet watch`` by simply adding ``watch`` after ``dotnet``. For example:
+
+- ``dotnet run``: ``dotnet watch run``
+- ``dotnet run --f net451``: ``dotnet watch run --f net451``
+- ``dotnet run --f net451 -- -arg1``: ``dotnet watch run -f net451 -- -arg1``
+- ``dotnet test``: ``dotnet watch test``
+
+To run ``WebApp`` using the watcher, run ``dotnet watch run`` in the ``WebApp`` folder. The console output will show messages similar to the ones below, indicating that ``dotnet watch`` is now watching code files:
+
+.. code-block:: bash
+
+  user$ dotnet watch run
+  [DotNetWatcher] info: Running dotnet with the following arguments: run
+  [DotNetWatcher] info: dotnet process id: 39746
+  Project WebApp (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+Making changes with ``dotnet watch``
+------------------------------------
+
+Let's fix the bug that we discovered when we tried to compute the product of two number.
+
+Open ``WebApp/Controllers/MathController.cs`` in a text editor.
+
+You'll notice that we've intentionally introduced a bug in the code.
+
+.. literalinclude:: dotnet-watch/sample/WebApp/Controllers/MathController.cs
+   :language: c#
+   :lines: 12-17
+   :emphasize-lines: 5
+   :dedent: 4
+
+Fix the code by replacing ``a + b`` with ``a * b``.
+
+Save the file. The console output will show messages similar to the ones below, indicating that ``dotnet watch`` detected a file change and restarted the application.
+
+
+.. code-block:: bash
+
+  [DotNetWatcher] info: File changed: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp/Controllers/MathController.cs
+  [DotNetWatcher] info: Running dotnet with the following arguments: run
+  [DotNetWatcher] info: dotnet process id: 39940
+  Project WebApp (.NETCoreApp,Version=v1.0) will be compiled because inputs were modified
+  Compiling WebApp for .NETCoreApp,Version=v1.0
+  Compilation succeeded.
+    0 Warning(s)
+    0 Error(s)
+  Time elapsed 00:00:03.3312829
+  
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+
+
+Refresh or open the webpage pointing to ``http://localhost:5000/api/math/product?a=4&b=5`` and you should get the correct result (20).
```

Comment:
Verify ``http://localhost:5000/api/math/product?a=4&b=5`` returns the correct result.
[don't need to tell them it's 20]